### PR TITLE
wfuzz: 2.4.2 -> 3.1.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 3.1.0

Changelog: https://github.com/xmendez/wfuzz/releases/tag/v3.1.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 107789` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.wfuzz</li>
    <li>wfuzz (python38Packages.wfuzz)</li>
    <li>python39Packages.wfuzz</li>
  </ul>
</details>

File: pkgs/development/python-modules/wfuzz/default.nix [hunk 1]
```diff
@@ -8,38 +9,49 @@
...
+    mock
+    future
+  ] ++ lib.optionals stdenv.hostPlatform.isWindows [
+    colorama
   ];
 
-  checkInputs = [ netaddr pytest ] ++ lib.optionals isPy27 [ mock ];
+  checkInputs = [
+    netaddr
+    pytest
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/wfuzz/default.nix:46]: Can we use pytestCheckHook here?

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107789` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.wfuzz</li>
    <li>wfuzz (python38Packages.wfuzz)</li>
    <li>python39Packages.wfuzz</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107789` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.wfuzz</li>
    <li>wfuzz (python38Packages.wfuzz)</li>
    <li>python39Packages.wfuzz</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 107789` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>python37Packages.wfuzz</li>
    <li>wfuzz (python38Packages.wfuzz)</li>
    <li>python39Packages.wfuzz</li>
  </ul>
</details>

File: pkgs/development/python-modules/wfuzz/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/python-modules/wfuzz/default.nix:47]: Updated

**Review by Pamplemousse — APPROVED** [review]
On NixOS: package builds, and binary runs (displays help).

**Comment by Pamplemousse** [comment]
@fabaff, maybe you could add yourself as a maintainer for this package.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107789` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
3 packages built:
</summary>
<ul>
<li>python37Packages.wfuzz</li>
<li>wfuzz (python38Packages.wfuzz)</li>
<li>python39Packages.wfuzz</li>
</ul>
</details>

**Pull request merged**